### DISCUSSION TITLE
Amazon Aurora - "P1011: Error opening a TLS connection" when using SSL (through SSH tunnel)

### DISCUSSION THREAD

**Discussion by phwt** [prisma/prisma#7002] [Q&A] [answered]
Hi, I have set up Amazon Aurora as listed below:

- Amazon Aurora (PostgreSQL) Cluster with 3 instances in each AZ
- All instances belong to a private subnet
- I have setup EC2 bastion host for use as SSH Tunnel

For local development I connect to ssh tunnel like this:

```zsh
ssh -N -L 5432:aurora-cluster.cluster-somerandomcharacter.ap-southeast-1.rds.amazonaws.com:5432 ec2-user@12.34.56.78 -i privateKeyOfEC2.cer
```

- `12.34.56.78` is the public IP of the bastion server

---

## Test Connection with `psql`

To verify this is not a problem with SSH tunnel. I have tested connecting to the cluster through SSH tunnel with `psql`

```zsh
psql postgresql://postgres@localhost:5432/DBNAME
```

And it does connect.

[![enter image description here][1]][1]

---

## Connect with Prisma

First I try to connect without SSL:

`.env`
```zsh
DATABASE_URL="postgresql://postgres:password@localhost:5432/DBNAME?schema=public"
```

I run `prisma migrate dev` as usual

```
Already in sync, no schema change or pending migration was found.

✔ Generated Prisma Client (2.20.1) to ./node_modules/@prisma/client in 113ms
```

And then I try to use Prisma Client.

```
Error: 
Invalid `prisma.work.create()` invocation:

  Authentication failed against database server at `localhost`, the provided database credentials for `postgres` are not valid.

Please make sure to provide valid database credentials for the database server at `localhost`.
```

The credentials are the same ones used when running `prisma migrate` (which succeeded) this left me confused but after some searching, I noticed this might be SSL-related issues.

---

## Connect with Prisma (again with SSL)

I have downloaded the certificate from [RDS SSL User Guide](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html) and place it in `./prisma`

`.env`
```zsh
DATABASE_URL="postgresql://postgres:password@localhost:5432/DBNAME?schema=public&sslmode=require&sslcert=rds-combined-ca-bundle.cer&sslaccept=accept_invalid_certs"
```

I tried running `prisma migrate dev` again

```zsh
Error: P1011: Error opening a TLS connection: One or more parameters passed to a function were not valid.
```

I have also tried using Prisma Client throws the same error as expected:

```zsh
Error: 
Invalid `prisma.work.create()` invocation:

  Error opening a TLS connection: One or more parameters passed to a function were not valid.
```

I have tried to set `sslaccept`, `sslmode` as in #5132 but still have the same problem.

---

## Additional Information

**Aurora Config**

- Instance class: `db.t3.medium`
- Engine version: `11.9`
- Public accessibility: `No`

**Prisma Version**

```zsh
❯ npx prisma -v
Environment variables loaded from .env
prisma               : 2.20.1
@prisma/client       : 2.20.1
Current platform     : darwin
Query Engine         : query-engine 60ba6551f29b17d7d6ce479e5733c70d9c00860e (at node_modules/@prisma/engines/query-engine-darwin)
Migration Engine     : migration-engine-cli 60ba6551f29b17d7d6ce479e5733c70d9c00860e (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine : introspection-core 60ba6551f29b17d7d6ce479e5733c70d9c00860e (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary        : prisma-fmt 60ba6551f29b17d7d6ce479e5733c70d9c00860e (at node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash : 60ba6551f29b17d7d6ce479e5733c70d9c00860e
Studio               : 0.365.0
```

  [1]: https://i.stack.imgur.com/onOcq.png

**Comment by janpio** [comment, marked as answer]
This sounds more like an issue than a discussion. Can you maybe recreate this as an issue? Will make it much easier for us to debug and help you. (Unfortunately we can not convert discussions to issues at Github.) Optimally really even two issues: One for the "without SSL" case and on for the "with SSL" one - seems like to very different problems.

**Reply by phwt** [reply]
My problem is similar to #5132 which is resolved. I'm not sure if this problem was caused by my own error. What do you suggest?

**Reply by janpio** [reply]
Create an issue, then we can figure that out. Better one issue to many than one to few. If it turns out we figure it out, we can just close the issue :D

**Reply by phwt** [reply]
Thanks, I have opened an issue at #7048.
